yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -177,14 +177,26 @@ package object config {
 
   private[spark] val DRIVER_CORES = ConfigBuilder("spark.driver.cores")
     .intConf
-    .optional
+    .withDefault(1)
 
   private[spark] val DRIVER_MEMORY_OVERHEAD = ConfigBuilder("spark.yarn.driver.memoryOverhead")
     .bytesConf(ByteUnit.MiB)
     .optional
 
+  private[spark] val DRIVER_MEMORY = ConfigBuilder("spark.driver.memory")
```

Comment:
This is a core config, so should be declared in `core/src/main/scala/org/apache/spark/internal/config/package.scala` instead. Same for `EXECUTOR_MEMORY`, `PY_FILES`, and probably also `EXECUTOR_CORES`, although not completely sure about that last one.